crates/ruff/src/rules/ruff/rules/type_alias_annotation.rs

```diff
@@ -0,0 +1,89 @@
+use ruff_python_ast::{Expr, ExprName, Ranged, Stmt, StmtAnnAssign, StmtTypeAlias};
+
+use crate::{registry::AsRule, settings::types::PythonVersion};
+use ruff_diagnostics::{AutofixKind, Diagnostic, Edit, Fix, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for use of `TypeAlias` annotation for declaring type aliases.
+///
+/// ## Why is this bad?
+/// The `type` keyword was introduced in Python 3.12 by PEP-695 for defining type aliases.
+/// The type keyword is easier to read and provides cleaner support for generics.
+///
+/// ## Example
+/// ```python
+/// ListOfInt: TypeAlias = list[int]
+/// ```
+///
+/// Use instead:
+/// ```python
+/// type ListOfInt = list[int]
+/// ```
+#[violation]
+pub struct TypeAliasAnnotation {
+    name: String,
+}
+
+impl Violation for TypeAliasAnnotation {
+    const AUTOFIX: AutofixKind = AutofixKind::Always;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TypeAliasAnnotation { name } = self;
+        format!("Type alias `{name}` uses `TypeAlias` annotation instead of the `type` keyword.")
```

Comment:
I think we tend not to add trailing punctuation unless the message spans more than one sentence.